hydroflow_datalog_core/src/grammar.rs

```diff
@@ -32,6 +31,13 @@ pub mod datalog {
         _dot: Option<()>,
     }
 
+    #[derive(Debug, Clone)]
+    pub enum RuleType {
+        Sync(#[rust_sitter::leaf(text = ":-")] ()),
+        NextTick(#[rust_sitter::leaf(text = ":+")] ()),
+        Async(#[rust_sitter::leaf(text = ":~")] ()),
```

Comment:
This is a new bloom-inspired syntax?

Reply:
Yep! Much simpler than having to analyze the flow of `t` through `succ`/`choose`.